src/overlays/actors/ovl_Obj_Rotlift/z_obj_rotlift.c

```diff
@@ -28,24 +35,109 @@ const ActorInit Obj_Rotlift_InitVars = {
     (ActorFunc)ObjRotlift_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B96178[] = {
+struct ModelInfo D_80B96160[] = {
+    {
+        object_rotlift_DL_000400,
+        object_rotlift_Matanimheader_001F98,
+        &object_rotlift_Colheader_002190,
+    },
+    {
+        object_rotlift_DL_002CE0,
+        object_rotlift_Matanimheader_004A08,
+        &object_rotlift_Colheader_004DF0,
+    },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 800, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 800, ICHAIN_STOP),
 };
 
-#endif
+void ObjRotlift_MoveDekuFlowers(ObjRotlift* this) {
+    ObjEtcetera** dekuFlower = this->dekuFlowers;
+    ObjEtcetera* curDekuFlower;
+    f32 posOffset = 300.0f;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->dekuFlowers); i++, dekuFlower++) {
+        curDekuFlower = *dekuFlower;
+        if (curDekuFlower->dyna.actor.update == NULL) {
+            *dekuFlower = NULL;
+        } else {
+            curDekuFlower->dyna.actor.world.pos.x =
+                this->dyna.actor.world.pos.x + posOffset * Math_SinS(this->dyna.actor.shape.rot.y);
+            curDekuFlower->dyna.actor.world.pos.y = this->dyna.actor.world.pos.y + 380.0f * this->dyna.actor.scale.y;
+            curDekuFlower->dyna.actor.world.pos.z =
+                this->dyna.actor.world.pos.z + posOffset * Math_CosS(this->dyna.actor.shape.rot.y);
+            curDekuFlower->dyna.actor.shape.rot.y = this->dyna.actor.shape.rot.y;
+        }
+        posOffset -= 600.0f;
+    }
+}
+
+void ObjRotlift_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    ObjRotlift* this = THIS;
+    s32 type;
+    s32 dekuFlowerType;
+    s32 i;
+    ModelInfo* sModelInfo;
+    ObjEtcetera** dekuFlowers;
+
+    type = OBJROTLIFT_GET_TYPE(&this->dyna.actor);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    if (type == 0) {
+        for (dekuFlowers = this->dekuFlowers, i = 0; i < 2; i++, dekuFlowers++) {
```

Comment:
```suggestion
        for (dekuFlowers = this->dekuFlowers, i = 0; i < ARRAY_COUNT(this->dekuFlowers); i++, dekuFlowers++) {
```
is a bit clearer and should give the same codegen.